sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -14,6 +15,75 @@ namespace sdk
 {
 namespace trace
 {
+/**
+ * A counterpart to AttributeValue that makes sure a value is owned. This
+ * replaces all non-owning references with owned copies.
+ */
+using SpanDataAttributeValue = nostd::variant<bool,
```

Comment:
My main concern with this approach that we could've possibly avoided adding a totally new class, if we implemented a new method on original collection that allows to create a clone / deep copy of the original `common::AttributeValue` s into another.

This could be solved by adding `nostd::string` and `const char *` on original `AttributeValue`. While performing a deep-copy of the container - we could copy the contents of `nostd::string_view` into `nostd::string` or maybe `nostd::shared_ptr<char[]> ptr(new char[size_])` + 0 byte? Or alike, something that can hold a string value copied from API surface, but still have the same class definition on API surface as inside the SDK.

There's something not entirely right with adding a totally new class for that.